src/wp-includes/media.php

```diff
@@ -5759,9 +5766,12 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 
 	/*
 	 * If flag was set based on contextual logic above, increase the content
-	 * media count if the image size is larger than the threshold.
+	 * media count, either unconditionally, or based on whether the image size
+	 * is larger than the threshold.
 	 */
-	if ( $maybe_increase_count ) {
+	if ( $increase_count ) {
```

Comment:
The pixel limitation was only added later, that said I am still not 100% sure we should add it here too just because we started using it. What is different between the images here and the ones where we only increase the counter based on minimum size is:
* The images here are part of post content, i.e. the main content of the page.
* The images where we only increase based on minimum size are supposedly in the header, but how "important" they are is unknown to us.

So there's a potential argument that images in post content should always be counted as the assumption is that they always carry some kind of significance. Which is potentially different for images in the header, which _could_ be a featured image, but could also be some tiny social media icon or the site logo or some ad graphic.

I don't disagree with you, maybe we should indeed increase the counter here conditionally as well (separate from this PR though). But I wanted to share the small nuance there is between the two situations.

Reply:
Thanks for the explanation. That makes sense to me, but does seem like something we might want to reconsider in a separate issue, given that the inconsistency. Definitely agree it's not something to change in this PR.